packages/manager/src/managers/screenshots/ScreenshotsManager.ts

```diff
@@ -253,6 +253,9 @@ export class ScreenshotsManager extends BaseManager {
 	async uploadScreenshot(
 		args: ScreenshotsManagerUploadScreenshotArgs,
 	): Promise<ScreenshotsManagerUploadScreenshotReturnType> {
+		if (this._s3ACL === undefined) {
+			await this.initS3ACL();
+		}
```

Comment:
I don't get why this need to appear here, the ACL is inited upon start and/or login of user. This could cause issues if the user is not logged in.

ACL assertion is made after to ensure the ACL is ready.

Reply:
Just to add some context, the reason we prefer to init the ACL context separately (similarly to the reason we init the plugin runner separately and not in each function), is because it makes the app flow much simpler to debug.

- ACL/plugin runner are like car engines, you start them once, and then leverage them to drive
   *In the case of the ACL, we start it once when we can (when user is logged in or just logged in), you start the car engine when you have the key for it.*
- Decoupling the engine start from engine usage allows for easier troubleshooting
   *If you were checking that the engine is started, start it if not, to switch gear, you wouldn't know if the car engine startup failed or the switch gear failed if both are done at the same time.*
   *That's why we just check that the engine is started with `assert...()` functions, so that if this fails, it indicates a clear fail in the way the manager is used (i.e. a bug in the way the code is written, not in the runtime).*
- Decoupling the engine start from engine usage allows for easier implementation of functions using the engine
   *To keep on going with the car analogy, you don't want to check if the engine is started, and start it if not, whenever you switch gears, accelerate, turn on the windscreen wiper, or turn on the lights. It's better to let it fail gracefully if so, hence the `assert...()` functions indicating a bug in the way the code is written.*